tools/sistr_cmd/sistr_cmd.xml

```diff
@@ -0,0 +1,227 @@
+<tool id="sistr_cmd" name="sistr_cmd" version="0.3.1">
+  <description>
+    Salmonella In Silico Typing Resource commandline tool for serovar prediction
+  </description>
+  <requirements>
+    <requirement type="package" version="0.3.1">sistr_cmd</requirement>
+  </requirements>
+  <stdio>
+    <exit_code range="1:" />
+  </stdio>
+  <command><![CDATA[
+  sistr 
+    -f $output_format
+    #if $output_format == "csv"
+      -o sistr-report.csv
+    #elif $output_format == "json"
+      -o sistr-report.json
+    #end if
+    -p $cgmlst_profiles
+    -n $novel_alleles
+    -a $alleles_output
+    -T $tmp_dir
+    #if $keep_tmp
+      -K
+    #end if
+    #if $use_full_cgmlst_db
+      --use-full-cgmlst-db
+    #end if
+    #if $no_cgmlst
+      --no-cgmlst
+    #end if
+    #if $run_mash
+      -m
+    #end if
+    #if $qc
+      --qc
+    #end if
+    #if $threads > 1
```

Comment:
Can you change to `"\${GALAXY_SLOTS:-1}"` which would read as "set to `$GALAXY_SLOTS`, or default to `1`"?

And yes, it's okay to create a multiprocessing threadpool, so long as it limits itself to the Galaxy slots define above.

Behind the scenes, this allows us to write a set of rules like

```yml
spades:
  rules:
      - rule_type: file_size
        nice_value: 0
        lower_bound: 10 MB
        upper_bound: 1 GB
        destination: slurm_8core_32G_high
```

That is, at runtime of a particular job we can measure total size of input files (or number of input files, or other parameters) and adjust resources (such as processing cores or memory) given to this job on our cluster based on that information (here we map spades jobs with input sizes from 10 MB to 1 GB to reserve 8 cores and 32 GB of memory on our cluster).